lightwood/encoder/categorical/onehot.py

```diff
@@ -1,140 +1,174 @@
+        :param target_weights: Percentage of total population represented by each category (between [0, 1]).
+        :param mode: True uses an extra dimension to account for unknown/out-of-distribution categories
+        """  # noqa
+
+        super().__init__(is_target)
+        self.map = None  # category name -> index
+        self.rev_map = None  # index -> category name
+        self.use_unknown = use_unknown
 
         self.target_weights = None
-        self.index_weights = None
+        self.inv_target_weights = None
         if self.is_target:
             self.target_weights = target_weights
 
-    def prepare(self, priming_data, max_dimensions=20000):
+    def prepare(self, priming_data: Iterable[str]):
+        """
+        Prepares the OHE Encoder by creating a dictionary mapping.
+
+        Unknown categories must be explicitly handled as python `None` types.
+        """
         if self.is_prepared:
             raise Exception('You can only call "prepare" once for a given encoder.')
 
-        self._lang = Lang('default')
-        if self.handle_unknown == "return_zeros":
-            priming_data = [x for x in priming_data if x is not None]
-            self._lang.index2word = {}
-            self._lang.word2index = {}
-            self._lang.n_words = 0
-        elif self.handle_unknown == "unknown_token":
-            priming_data = [x if x is not None else UNCOMMON_WORD for x in priming_data]
-            self._lang.index2word = {UNCOMMON_TOKEN: UNCOMMON_WORD}
-            self._lang.word2index = {UNCOMMON_WORD: UNCOMMON_TOKEN}
-            self._lang.word2count[UNCOMMON_WORD] = 0
-            self._lang.n_words = 1
-
-        for category in priming_data:
-            if category is not None:
-                self._lang.addWord(str(category))
-
-        while self._lang.n_words > max_dimensions:
-            if self.handle_unknown == "return_zeros":
-                necessary_words = []
-            elif self.handle_unknown == "unknown_token":
-                necessary_words = [UNCOMMON_WORD]
-            least_occuring_words = self._lang.getLeastOccurring(n=len(necessary_words) + 1)
-
-            word_to_remove = None
-            for word in least_occuring_words:
-                if word not in necessary_words:
-                    word_to_remove = word
-                    break
-
-            self._lang.removeWord(word_to_remove)
-
-        # Note: Is target assume that we are operating in "unknown_token" mode
+        unq_cats = np.unique([i for i in priming_data if i is not None]).tolist()
+        if self.use_unknown:
+            log.info("Encoding UNK categories as index 0")
+            self.map = {cat: indx + 1 for indx, cat in enumerate(unq_cats)}
+            self.map.update({_UNCOMMON_WORD: 0})
+            self.rev_map = {indx: cat for cat, indx in self.map.items()}
+        else:
+            log.info("Encoding UNK categories as vector of all 0s")
```

Comment:
probably better to say `unkown` since `UNK` might confuse users and idk if there's a reason to use that abbreviation. Also `unkown categories and missing values` might be more explicit as to what this is doing.

Reply:
will use `unknown` albeit in caps

Reply:
added